sui_core/src/gateway_state.rs

```diff
@@ -573,7 +575,19 @@ where
         tx: Transaction,
     ) -> Result<TransactionResponse, anyhow::Error> {
         let tx_kind = tx.data.kind.clone();
-        let (certificate, effects) = self.execute_transaction_impl(tx).await?;
+        let mut res = self.execute_transaction_impl(tx.clone()).await;
+
+        let mut remaining_retries = MAX_NUM_TX_RETRIES - 1;
+        while res.is_err() {
+            if remaining_retries == 0 {
+                res?;
```

Comment:
I think `return res;` would be more explicit.

Reply:
This wont really work.
`res` is of type `Result<(CertifiedTransaction, TransactionEffects), anyhow::Error>`
fn expects `Result<TransactionResponse, anyhow::Error>`

Instead I'll use `unwrap_err`